cts-java/src/test/java/org/ga4gh/cts/api/reads/ReadsSearchIT.java

```diff
@@ -0,0 +1,263 @@
+
+        // get a ReadGroup id
+        final String readGroupId = Utils.getReadGroupId(client);
+
+        final SearchReadsRequest request =
+                SearchReadsRequest.newBuilder()
+                                  .setReadGroupIds(aSingle(readGroupId))
+                                  .setStart(0L)
+                                  .setEnd(150L)
+                                  .setReferenceId(refId)
+                                  .build();
+        // XXX fails due to server issue #591
+        final SearchReadsResponse response = client.reads.searchReads(request);
+
+        assertThat(response.getAlignments()).isNotNull();
+
+        response.getAlignments().stream()
+                .forEach(readAlignment ->
+                                 assertThat(readAlignment.getAlignedSequence()).isNotNull()
+                                                                               .matches("[ACTGN]+"));
+    }
+
+    /**
+     * Verify that passing all known read group names in a {@link SearchReadsRequest}
+     * returns all matching read groups.
+     * XXX fails due to server issue #596
+     *
+     * @throws Exception if there's a problem
+     */
+    @Test
+    public void searchReadsWithAllIdsReturnsReadsForEach() throws Exception {
+        // first get a valid reference
+        final String refId = Utils.getValidReferenceId(client);
+
+        // get all ReadGroupSets
+        final List<ReadGroupSet> allReadGroupSets = Utils.getAllReadGroupSets(client);
+
+        // collect all IDs from the ReadGroups
+        final List<String> allReadGroupIds =
+                allReadGroupSets.stream()
+                                .flatMap(readGroupSet ->
+                                                 readGroupSet.getReadGroups()
+                                                             .stream())
+                                .map(ReadGroup::getId)
+                                .collect(Collectors.toList());
+
+        final SearchReadsRequest request =
+                SearchReadsRequest.newBuilder()
+                                  .setReadGroupIds(allReadGroupIds)
+                                  .setStart(0L)
+                                  .setEnd(150L)
+                                  .setReferenceId(refId)
+                                  .build();
+        final SearchReadsResponse response = client.reads.searchReads(request);
+
+        assertThat(response.getAlignments()).isNotNull();
+
+        response.getAlignments().stream()
+                .forEach(readAlignment ->
+                                 assertThat(readAlignment.getAlignedSequence()).isNotNull()
+                                                                               .matches("[ACTGN]+"));
+    }
+
+    /**
+     * <p>Verify aligned sequences contain only the permitted symbols.</p>
+     * <p>In any {@link ReadAlignment}, the <tt>alignedSequence</tt> field can only contain
```

Comment:
There is no such requirement in the schemas, as this comment appears to imply. However, the test suite controls the input data, so you could assert this as a property of the input data.